src/main/java/org/craftercms/studio/controller/rest/v2/CmisController.java

```diff
@@ -111,15 +119,42 @@ public ResponseBody cloneContent(@RequestBody CmisCloneRequest cmisCloneRequest)
         return responseBody;
     }
 
-    @PostMapping(value = "/api/2/cmis/upload", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
-    public ResponseBody uploadContent(@RequestParam(value = "siteId", required = true) String siteId,
-                                      @RequestParam(value = "cmisRepoId", required = true) String cmisRepoId,
-                                      @RequestParam(value = "cmisPath", required = true) String cmisPath,
-                                      @RequestParam(value = "file", required = true) MultipartFile file)
+    @PostMapping(value = "/api/2/cmis/upload")
+    public ResponseBody uploadContent(HttpServletRequest httpServletRequest)
             throws IOException, CmisUnavailableException, CmisPathNotFoundException, CmisTimeoutException,
-            CmisRepositoryNotFoundException {
-        cmisService.uploadContent(siteId, cmisRepoId, cmisPath, file.getOriginalFilename(), file.getInputStream());
-
+            CmisRepositoryNotFoundException, FileUploadException {
+
+        ServletFileUpload servletFileUpload = new ServletFileUpload();
+        FileItemIterator itemIterator = servletFileUpload.getItemIterator(httpServletRequest);
+        String filename = StringUtils.EMPTY;
+        String siteId = StringUtils.EMPTY;
+        String cmisRepoId = StringUtils.EMPTY;
+        String cmisPath = StringUtils.EMPTY;
+        while (itemIterator.hasNext()) {
+            FileItemStream item = itemIterator.next();
+            String name = item.getFieldName();
+            try (InputStream stream = item.openStream()) {
+                if (item.isFormField()) {
+                    switch (name) {
+                        case REQUEST_PARAM_SITEID:
```

Comment:
Maybe be should validate if any of the parameters is missing and throw an exception